app/commands/source/add_value_command.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module Source
+  class AddValueCommand < ApplicationCommand
+    usage do
+      no_command
+    end
+
+    keyword :name do
+      desc "Name for the new value"
+      required
+    end
+
+    keyword :attribute_friendly_id do
+      desc "Friendly ID of the primary attribute to add the value to"
+      required
+    end
+
+    def execute
+      frame("Adding new value") do
+        ActiveRecord::Base.transaction do
+          setup!
+          create_value!
+          create_attributes_value!
+        rescue ActiveRecord::Rollback => e
+          logger.fatal(e.message)
+          exit(1)
+        end
+
+        update_data_files!
+      end
+    end
+
+    private
+
+    def setup!
+      @primary_attribute = Attribute.find_by(friendly_id: params[:attribute_friendly_id])
+
+      raise ActiveRecord::Rollback,
+        "Primary attribute `#{params[:attribute_friendly_id]}` not found" if @primary_attribute.nil?
+
+      @extended_attributes = @primary_attribute.extended_attributes
+      @manually_sorted = @primary_attribute.manually_sorted?
+    end
+
+    def create_value!
+      value_identifier = "#{@primary_attribute.handle}__#{params[:name]}"
+      friendly_id = StringTransformer.generate_friendly_id(value_identifier)
+
+      existing_value = Value.find_by(friendly_id: friendly_id)
+
+      raise ActiveRecord::Rollback,
+        "Value `#{friendly_id}` already exists" if existing_value
+
+      position = @primary_attribute.values.map(&:position).compact.max + 1 if @manually_sorted
```

Comment:
This also feels like it should be handled by the model vs in here?

Also when we write it, I think you can use `max_by`?
```suggestion
      position = @primary_attribute.values.max_by(&:position) + 1 if @manually_sorted
```